docs/addons/karpenter.md

```diff
@@ -32,12 +30,12 @@ import * as blueprints from '@aws-quickstart/eks-blueprints';
 const app = new cdk.App();
 
 const karpenterAddonProps = {
-  provisionerSpecs: {
-    'node.kubernetes.io/instance-type': ['m5.2xlarge'],
-    'topology.kubernetes.io/zone': ['us-east-1c'],
-    'kubernetes.io/arch': ['amd64','arm64'],
-    'karpenter.sh/capacity-type': ['spot','on-demand'],
-  },
+  requirements: [
```

Comment:
Is it a breaking change, e.g if customer already using Karpenter previous version, will this break it?

Reply:
If they are using the previous addon version, this will break it. I can add notes to ensure the changes and an upgrade path, although I'm not sure it is sustainable to keep documenting every breaking change.

Reply:
Upgrade path could be 1/ remove add-on and deploy 2/ add the add-on back and deploy. However, I am unsure what operation impact is going to be when karpenter is removed: I hope all nodes will be running as is.

Reply:
If the addon remove the provisioner karpenter will remove the nodes

Reply:
@youngjeong46 based on the above, we need an upgrade path.

Reply:
would it suffice to document the changes in the "requirement" prop? The values would remain the same, it's just the prop type that changed. 

As for version upgrades specifics (like 0.14 -> 0.15 or 0.15 -> 0.16) is documented in the official Karpenter site which is listed under troubleshooting at the bottom.

Reply:
Yes, and let's document and mention in the PR description so that it gets to the release notes, e.g., in the description of changes adding a bullet about upgrade path documentation